baseplate/clients/thrift.py

```diff
@@ -213,38 +214,38 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                         prot.trans.set_header(b"Edge-Request", edge_context)
 
                     result = method(*args, **kwargs)
-            except TTransportException as exc:
-                # the connection failed for some reason, retry if able
-                span.finish(exc_info=sys.exc_info())
-                last_error = str(exc)
-                if exc.inner is not None:
-                    last_error += f" ({exc.inner})"
-                continue
-            except (TApplicationException, TProtocolException):
-                # these are subclasses of TException but aren't ones that
-                # should be expected in the protocol. this is an error!
-                span.finish(exc_info=sys.exc_info())
-                raise
-            except Error as exc:
-                # a 5xx error is an unexpected exception but not 5xx are
-                # not.
-                if 500 <= exc.code < 600:
+                except TTransportException as exc:
+                    # the connection failed for some reason, retry if able
+                    span.finish(exc_info=sys.exc_info())
+                    last_error = str(exc)
+                    if exc.inner is not None:
+                        last_error += f" ({exc.inner})"
+                    continue
+                except (TApplicationException, TProtocolException):
+                    # these are subclasses of TException but aren't ones that
+                    # should be expected in the protocol. this is an error!
+                    span.finish(exc_info=sys.exc_info())
+                    raise
+                except Error as exc:
+                    # a 5xx error is an unexpected exception but not 5xx are
```

Comment:
What else would trigger this codepath if not a 5xx error? should this end in "raise" if it's not an error?

Reply:
It don't want to change the behavior here, it seems it's expected that the user handle any thrift errors. Why we discard the exception information in the span if it's not 5xx I don't know, the comment seems to indicate it's not interesting? (it's a confusing comment)

Reply:
Seemingly correct when the endpoint is down:

```
✦3 ❯ kubectl exec -it service/reddit-service-alt-text -- ./test-client example
Traceback (most recent call last):
  File "/src/alt_text/alt_text_thrift/AltTextService-remote", line 119, in <module>
    pp.pprint(client.example())
  File "/src/alt_text/alt_text_thrift/AltTextService.py", line 73, in example
    return self.recv_example()
  File "/src/alt_text/alt_text_thrift/AltTextService.py", line 84, in recv_example
    (fname, mtype, rseqid) = iprot.readMessageBegin()
  File "/usr/local/lib/venv/lib/python3.9/site-packages/thrift/protocol/TBinaryProtocol.py", line 134, in readMessageBegin
    sz = self.readI32()
  File "/usr/local/lib/venv/lib/python3.9/site-packages/thrift/protocol/TBinaryProtocol.py", line 217, in readI32
    buff = self.trans.readAll(4)
  File "/usr/local/lib/venv/lib/python3.9/site-packages/thrift/transport/TTransport.py", line 62, in readAll
    chunk = self.read(sz - have)
  File "/usr/local/lib/venv/lib/python3.9/site-packages/thrift/transport/TTransport.py", line 164, in read
    self.__rbuf = BufferIO(self.__trans.read(max(sz, self.__rbuf_size)))
  File "/usr/local/lib/venv/lib/python3.9/site-packages/thrift/transport/TSocket.py", line 166, in read
    raise TTransportException(type=TTransportException.END_OF_FILE,
thrift.transport.TTransport.TTransportException: TSocket read 0 bytes
command terminated with exit code 1
```

Reply:
Apparently, we treat some of these differently because apparently we consider "Expected" exceptions (like hitting a 4xx, which is the client doing something bad (usually at least...) and other "non server" exceptions to not be failures. So they're not raised/handled the same way.